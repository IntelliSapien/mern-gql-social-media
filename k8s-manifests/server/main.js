"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.MyChart = void 0;
const cdk8s_1 = require("cdk8s");
const cdk8s_debore_1 = require("cdk8s-debore");
class MyChart extends cdk8s_1.Chart {
    constructor(scope, id, props = {}) {
        super(scope, id, props);
        new cdk8s_debore_1.DeboredApp(this, "mern-gql-server", {
            image: "abhiit89/mern-gql:latest",
            autoScale: true,
            defaultReplicas: 1,
            namespace: "mern-gql",
            ingress: cdk8s_debore_1.IngressType.CLUSTER_IP,
            containerPort: 4000,
        });
    }
}
exports.MyChart = MyChart;
const app = new cdk8s_1.App();
new MyChart(app, "mern-gql-server");
app.synth();
//# sourceMappingURL=data:application/json;base64,eyJ2ZXJzaW9uIjozLCJmaWxlIjoibWFpbi5qcyIsInNvdXJjZVJvb3QiOiIiLCJzb3VyY2VzIjpbIm1haW4udHMiXSwibmFtZXMiOltdLCJtYXBwaW5ncyI6Ijs7O0FBQ0EsaUNBQStDO0FBQy9DLCtDQUF1RDtBQUN2RCxNQUFhLE9BQVEsU0FBUSxhQUFLO0lBQ2hDLFlBQVksS0FBZ0IsRUFBRSxFQUFVLEVBQUUsUUFBb0IsRUFBRTtRQUM5RCxLQUFLLENBQUMsS0FBSyxFQUFFLEVBQUUsRUFBRSxLQUFLLENBQUMsQ0FBQztRQUV4QixJQUFJLHlCQUFVLENBQUMsSUFBSSxFQUFFLGlCQUFpQixFQUFFO1lBQ3RDLEtBQUssRUFBRSwwQkFBMEI7WUFDakMsU0FBUyxFQUFFLElBQUk7WUFDZixlQUFlLEVBQUUsQ0FBQztZQUNsQixTQUFTLEVBQUUsVUFBVTtZQUNyQixPQUFPLEVBQUUsMEJBQVcsQ0FBQyxVQUFVO1lBQy9CLGFBQWEsRUFBRSxJQUFJO1NBQ3BCLENBQUMsQ0FBQztJQUNMLENBQUM7Q0FDRjtBQWJELDBCQWFDO0FBRUQsTUFBTSxHQUFHLEdBQUcsSUFBSSxXQUFHLEVBQUUsQ0FBQztBQUN0QixJQUFJLE9BQU8sQ0FBQyxHQUFHLEVBQUUsaUJBQWlCLENBQUMsQ0FBQztBQUNwQyxHQUFHLENBQUMsS0FBSyxFQUFFLENBQUMiLCJzb3VyY2VzQ29udGVudCI6WyJpbXBvcnQgeyBDb25zdHJ1Y3QgfSBmcm9tIFwiY29uc3RydWN0c1wiO1xuaW1wb3J0IHsgQXBwLCBDaGFydCwgQ2hhcnRQcm9wcyB9IGZyb20gXCJjZGs4c1wiO1xuaW1wb3J0IHsgRGVib3JlZEFwcCwgSW5ncmVzc1R5cGUgfSBmcm9tIFwiY2RrOHMtZGVib3JlXCI7XG5leHBvcnQgY2xhc3MgTXlDaGFydCBleHRlbmRzIENoYXJ0IHtcbiAgY29uc3RydWN0b3Ioc2NvcGU6IENvbnN0cnVjdCwgaWQ6IHN0cmluZywgcHJvcHM6IENoYXJ0UHJvcHMgPSB7fSkge1xuICAgIHN1cGVyKHNjb3BlLCBpZCwgcHJvcHMpO1xuXG4gICAgbmV3IERlYm9yZWRBcHAodGhpcywgXCJtZXJuLWdxbC1zZXJ2ZXJcIiwge1xuICAgICAgaW1hZ2U6IFwiYWJoaWl0ODkvbWVybi1ncWw6bGF0ZXN0XCIsXG4gICAgICBhdXRvU2NhbGU6IHRydWUsXG4gICAgICBkZWZhdWx0UmVwbGljYXM6IDEsXG4gICAgICBuYW1lc3BhY2U6IFwibWVybi1ncWxcIixcbiAgICAgIGluZ3Jlc3M6IEluZ3Jlc3NUeXBlLkNMVVNURVJfSVAsXG4gICAgICBjb250YWluZXJQb3J0OiA0MDAwLFxuICAgIH0pO1xuICB9XG59XG5cbmNvbnN0IGFwcCA9IG5ldyBBcHAoKTtcbm5ldyBNeUNoYXJ0KGFwcCwgXCJtZXJuLWdxbC1zZXJ2ZXJcIik7XG5hcHAuc3ludGgoKTtcbiJdfQ==