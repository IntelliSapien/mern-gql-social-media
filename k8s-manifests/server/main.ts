import { Construct } from "constructs";
import { App, Chart, ChartProps } from "cdk8s";
import { DeboredApp, IngressType } from "cdk8s-debore";
export class MyChart extends Chart {
  constructor(scope: Construct, id: string, props: ChartProps = {}) {
    super(scope, id, props);

    new DeboredApp(this, "mern-gql-server", {
      image: "abhiit89/mern-gql:latest",
      autoScale: true,
      defaultReplicas: 1,
      namespace: "mern-gql",
      ingress: IngressType.CLUSTER_IP,
      containerPort: 4000,
    });
  }
}

const app = new App();
new MyChart(app, "mern-gql-server");
app.synth();
