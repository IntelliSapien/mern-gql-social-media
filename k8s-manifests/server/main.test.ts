import {MyChart} from './main';
import {Testing} from 'cdk8s';

describe('mern-gql-server-config', () => {
  test('Generate mern-gql-server-config', () => {
    const app = Testing.app();
    const chart = new MyChart(app, "mern-gql-server");
    const results = Testing.synth(chart)
    expect(results).toMatchSnapshot();
  });
});
