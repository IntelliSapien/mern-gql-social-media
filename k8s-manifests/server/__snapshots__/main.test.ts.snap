// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Placeholder Empty 1`] = `Array []`;

exports[`mern-gql-server-config Generate mern-gql-server-config 1`] = `
Array [
  Object {
    "apiVersion": "apps/v1",
    "kind": "Deployment",
    "metadata": Object {
      "name": "mern-gql-server-deployment-c8f74d56",
      "namespace": "mern-gql",
    },
    "spec": Object {
      "selector": Object {
        "matchLabels": Object {
          "app": "merngqlserver7627D0B4",
        },
      },
      "template": Object {
        "metadata": Object {
          "labels": Object {
            "app": "merngqlserver7627D0B4",
          },
        },
        "spec": Object {
          "containers": Array [
            Object {
              "image": "abhiit89/mern-gql:latest",
              "imagePullPolicy": "Always",
              "name": "app",
              "ports": Array [
                Object {
                  "containerPort": 8080,
                },
              ],
              "resources": Object {
                "limits": Object {
                  "cpu": "400m",
                  "memory": "512Mi",
                },
                "requests": Object {
                  "cpu": "200m",
                  "memory": "256Mi",
                },
              },
            },
          ],
        },
      },
    },
  },
  Object {
    "apiVersion": "autoscaling/v2beta2",
    "kind": "HorizontalPodAutoscaler",
    "metadata": Object {
      "name": "mern-gql-server-deployment-hpa-c893cb13",
      "namespace": "mern-gql",
    },
    "spec": Object {
      "maxReplicas": 10,
      "metrics": Array [
        Object {
          "resource": Object {
            "name": "cpu",
            "target": Object {
              "averageUtilization": 85,
              "type": "Utilization",
            },
          },
          "type": "Resource",
        },
        Object {
          "resource": Object {
            "name": "memory",
            "target": Object {
              "averageUtilization": 75,
              "type": "Utilization",
            },
          },
          "type": "Resource",
        },
      ],
      "minReplicas": 1,
      "scaleTargetRef": Object {
        "apiVersion": "apps/v1",
        "kind": "Deployment",
        "name": "mern-gql-server-deployment-c8f74d56",
      },
    },
  },
  Object {
    "apiVersion": "v1",
    "kind": "Service",
    "metadata": Object {
      "name": "mern-gql-server-exposable-service-c8ae1813",
      "namespace": "mern-gql",
    },
    "spec": Object {
      "ports": Array [
        Object {
          "port": 80,
          "targetPort": 8080,
        },
      ],
      "selector": Object {
        "app": "merngqlserver7627D0B4",
      },
      "type": "ClusterIP",
    },
  },
]
`;
